iothub/device/src/Transport/Amqp/AmqpConnectionPool.cs

```diff
@@ -99,8 +84,17 @@ public void RemoveAmqpUnit(AmqpUnit amqpUnit)
                 {
                     AmqpConnectionHolder[] amqpConnectionHolders = ResolveConnectionGroup(deviceIdentity);
                     amqpConnectionHolder = ResolveConnectionByHashing(amqpConnectionHolders, deviceIdentity);
+
+                    amqpConnectionHolder.RemoveAmqpUnit(amqpUnit);
+
+                    // If the connection holder does not have any more units, the entry needs to be nullified.
+                    if (amqpConnectionHolder.IsEmpty())
+                    {
+                        int index = GetDeviceIdentityIndex(deviceIdentity, amqpConnectionHolders.Length);
+                        amqpConnectionHolders[index] = null;
+                        amqpConnectionHolder?.Dispose();
```

Comment:
Ok, yeah this is what I was wondering this morning on the call. So we do have to dispose this.